### ISSUE TITLE
Enhancement: MissingScopedDependencyQualifier should allow one unqualified + N qualified definitions

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#14]
## Summary

The `MissingScopedDependencyQualifier` rule reports a violation when there are multiple definitions of the same type, even when only one is unqualified (acting as a "default") and the rest use `named()` qualifiers. This is a valid Koin pattern that doesn't cause `DefinitionOverrideException`.

## Current Behavior

```kotlin
val networkModule = module {
    // Default HttpClient (no qualifier) — general purpose
    single { HttpClient(CIO) { /* default config */ } }

    // Specific HttpClient with qualifier — for a particular service
    single(named("apiClient")) { HttpClient(CIO) { /* API-specific config */ } }
}
```

**Detekt output:**
```
MissingScopedDependencyQualifier - Multiple HttpClient definitions without named() qualifiers
```

## Expected Behavior

This is debatable — ideally both should have qualifiers for clarity. However, the pattern of "one default + N qualified" is valid in Koin and doesn't cause runtime errors.

Possible approaches:
1. **Option A**: Only flag when there are 2+ definitions **without** qualifiers (actual ambiguity)
2. **Option B**: Keep current behavior but add a config option like `allowOneDefault: true`

## Impact

1 false positive in a real project.

## Environment

- Koin-Detekt 0.4.0
- Detekt 1.23.8
